test/export/test_export.py

```diff
@@ -526,6 +526,24 @@ def forward(self, x):
         ):
             torch.export.export(M(), (torch.randn(2, 3),), strict=False)
 
+    def test_state_primitives(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x = 1
+                self.y = {"x": 2}
+
+            def forward(self, x):
+                self.x = self.x + 3
+                self.y["x"] = self.y["x"] + 4
+                return x + self.x + self.y["x"]
```

Comment:
This should now generate code of the form `x + 4 + 6` (where 4 is inlined `self.x` and 6 is inlined `self.y["x"]`) but without this PR the generated code would be of the form `x + 7 + 10`.